test/main/one-choice.t

```diff
@@ -5,7 +5,8 @@ It doesn't present packages when there's only one choice
 
   $ touch "$TMP/foo.pkg"
   > search_packages() { printf "%s\n" "$TMP/foo.pkg"; }
```

Comment:
The `vercmp` fixture issue arose from this line in `test/main/one-choice.t` and from line 7 in `test/main/success.t`.

Because of `search_packages() { printf "%s\n" "$TMP/foo.pkg"; }`, a randomly created `$TMP` string from `cram` would be fed in to `main`, which would ultimately trickle down as an argument of the `act-like` version of `vercmp`. Since the arguments to this mocked `vercmp` will always have randomly generated prefix, it would always search for a different `md5sum` and would never find it, and that's why it would keep creating it.

I added a commit to fix this and it seems to work fine now. Would be great if you could test it on your system as well.

Reply:
I see, that mostly makes sense and the fix looks good. I'm a little concerned we're still missing something though.

The real `filter_packages` function has a condition for empty `operator` and `version` and should just `return 0` immediately (without calling `vercmp`) if they are empty. These tests call `main foo`, so why is this condition not being met?

Also, the one test you changed (`one-choice.t`) is actually dependent on how `filter_packages` behaves, so I'm worried that mocking it out like that is risky.

Reply:
And a minor note for next time: it'd be good to always commit any fixture changes in their own commit. GitHub was struggling to render that diff :joy:

Reply:
> The real filter_packages function has a condition for empty operator and version and should just return 0 immediately (without calling vercmp) if they are empty. These tests call main foo, so why is this condition not being met?

That's a really good point. Just looked and saw that the conditional check for no operator or version is inside `matches_name_version_filter`, which means the `vercmp` is already executed one line before it. Maybe we move the conditional check for no operator or version one level up to `filter_packages`?

Reply:
> And a minor note for next time: it'd be good to always commit any fixture changes in their own commit. GitHub was struggling to render that diff

Aha yes sorry about that.

Reply:
I reverted your fix and applied mine, WDYT?

Reply:
Oh shit, I just applied a fix and did not see this message, so maybe it overrode it. Maybe force push it?

Reply:
And another thing, I noticed that the phony `locale` is no longer present in the `Makefile`. Could we add the following line to `Makefile`:

```
.PHONY: locale
locale: locale/downgrade.pot
```

This just makes it easier to locally update the locale, and is also consistent with how we did it for `make man`. I will not change anything in case I conflict with your commits.